### DISCUSSION TITLE
Debug apps running on remote server

### DISCUSSION THREAD

**Discussion by stodolapatrik** [spatie/ray#422] [Q&A]
Hello, I have trouble with using Ray via SSH. (Laravel 8)
On localhost is it working perfectly, but on the ssh connection not. 

Do I set all things right ? 

What I did: 
- composer require spatie/laravel-ray
- php artisan ray:publish-config
- into app.php I added - \Spatie\LaravelRay\RayServiceProvider::class,
- into index() function which showing Index page I added ray('test message');

SSH connection succeed, and is located to www folder of current Laravel project

**Comment by stodolapatrik** [comment]
My hosting provider send me this - The SSH server does not run on the same machine as PHP, so it will not work for you. 

Is that the reason why Ray do not work?

**Reply by freekmurze** [reply]
That's probably the reason, yes.

**Reply by stodolapatrik** [reply]
And what is solution? Is the any other way how can I connect with Ray ?

**Reply by freekmurze** [reply]
The ray function in the PHP package send messages to a certain port (23517). The Ray desktop app listens for messages coming in via SSH on that port. You'll need to instruct your hostess to rely messages to port 23517 on the PHP server to the SSH server, so that it can transfer those message to the Ray client. 

I think there's a very low chance that your foster will do this.

**Reply by stodolapatrik** [reply]
And without SSH? That is not probably possible, right ?

**Reply by freekmurze** [reply]
Not possible as it would not be secure.
